### PR TITLE
Replace assert_eq! with similar_asserts::assert_eq!

### LINKED ISSUE #1362: [Feature] Replace `assert_eq!` with `similar_asserts::assert_eq!`
### Component

other

### Describe the feature you would like

`similar_asserts::assert_eq!` highlights the exacts differences between test subjects. This is incredibly helpful while debugging tests OR when we have to change test suites due to a breaking change e.g. https://github.com/alloy-rs/alloy/pull/1358

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xriazaka**
## Motivation
similar_asserts::assert_eq! highlights the exact differences between test subjects
Closes #1362 

## Solution
Replace assert_eq! with similar_asserts::assert_eq! on the "other" component

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

File: crates/serde/Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ workspace = true
 alloy-primitives = { workspace = true, features = ["rlp", "serde"] }
 serde.workspace = true
 serde_json = { workspace = true, features = ["alloc"] }
+similar-asserts.workspace = true
```
Comment (mattsse) [inline crates/serde/Cargo.toml:25]: this should actually be moved to dev-deps

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
we def also want this in more tests in the rpc crates

**Comment by 0xriazaka** [comment]
Sure will do that > we def also want this in more tests in the rpc crates

File: crates/rpc-client/src/builtin.rs
Comment (mattsse) [inline crates/rpc-client/src/builtin.rs:1]: these don't make much sense, similar_asserts is mainly useful for checking serde json, so lets limit the scope of this pr to that

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc-client/src/builtin.rs
Comment (0xriazaka) [reply crates/rpc-client/src/builtin.rs:1]: check #1433

**Comment by 0xriazaka** [comment]
resolved https://github.com/alloy-rs/alloy/pull/1433#discussion_r1788825627

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**